internal/cluster/clusterworkflow/activity_create_node_pool.go

```diff
@@ -0,0 +1,238 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusterworkflow
+
+import (
+	"context"
+
+	"emperror.dev/errors"
+	"github.com/jinzhu/gorm"
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/banzaicloud/pipeline/internal/cluster"
+	"github.com/banzaicloud/pipeline/internal/cluster/distribution"
+	eksworkflow "github.com/banzaicloud/pipeline/internal/providers/amazon/eks/workflow"
+	"github.com/banzaicloud/pipeline/pkg/cadence"
+	"github.com/banzaicloud/pipeline/pkg/providers"
+	"github.com/banzaicloud/pipeline/src/cluster/nodelabels"
+	"github.com/banzaicloud/pipeline/src/model"
+)
+
+const CreateNodePoolActivityName = "create-node-pool"
+
+type CreateNodePoolActivity struct {
+	clusters          cluster.Store
+	db                *gorm.DB
+	nodePools         cluster.NodePoolStore
+	eksNodePools      distribution.EKSNodePoolStore
+	awsSessionFactory AWSSessionFactory
+}
+
+// NewCreateNodePoolActivity returns a new CreateNodePoolActivity.
+func NewCreateNodePoolActivity(
+	clusters cluster.Store,
+	db *gorm.DB,
+	nodePools cluster.NodePoolStore,
+	eksNodePools distribution.EKSNodePoolStore,
+	awsSessionFactory AWSSessionFactory,
+) CreateNodePoolActivity {
+	return CreateNodePoolActivity{
+		clusters:          clusters,
+		db:                db,
+		nodePools:         nodePools,
+		eksNodePools:      eksNodePools,
+		awsSessionFactory: awsSessionFactory,
+	}
+}
+
+type CreateNodePoolActivityInput struct {
+	ClusterID   uint
+	UserID      uint
+	RawNodePool cluster.NewRawNodePool
+}
+
+func (a CreateNodePoolActivity) Execute(ctx context.Context, input CreateNodePoolActivityInput) error {
+	c, err := a.clusters.GetCluster(ctx, input.ClusterID)
+	if err != nil {
+		return cadence.WrapClientError(err)
+	}
+
+	switch {
+	case c.Cloud == providers.Amazon && c.Distribution == "eks":
```

Comment:
```suggestion
	case c.Cloud == providers.Amazon && c.Distribution == cluster.EKS:
```

there are constants defined for distributions in `pkg/cluster/base.go`

Reply:
I don't want to import that package here for now.